apps/console/src/features/identity-providers/components/utils/saml-idp-utils.ts

```diff
@@ -0,0 +1,304 @@
+                return "http://www.w3.org/2001/04/xmlenc#ripemd160";
+            case "SHA1":
+                return "http://www.w3.org/2000/09/xmldsig#sha1";
+            case "SHA256":
+                return "http://www.w3.org/2001/04/xmlenc#sha256";
+            case "SHA384":
+                return "http://www.w3.org/2001/04/xmldsig-more#sha384";
+            case "SHA512":
+                return "http://www.w3.org/2001/04/xmlenc#sha512";
+        }
+    };
+
+    if (property) {
+        return [ ...property.options.map((option: string, index: number) => {
+            return {
+                key: index,
+                text: displayNameMapper(option),
+                value: option
+            };
+        }), {
+            key: -1,
+            text: '',
+            value: ''
+        } ];
+    }
+
+    return [];
+
+};
+
+type FindPropValArgs<DefaultType> = { key: string; defaultValue: DefaultType; };
+type FindPropValFunction = <Type>(args: FindPropValArgs<Type>) => Type;
+
+type FindMetaArgs = { key: string };
+type FindMetaFunction = (args: FindMetaArgs) => CommonPluggableComponentMetaPropertyInterface;
+
+/**
+ * tldr: Pre computes idp metadata and data properties to find
+ * and return values in constant time.
+ *
+ * Why? to do a faster search we need a constant access time data structure to
+ * find the current value by key. This is because `data.properties` is a array of
+ * object { key, value } pairs which makes find operation exhaustive (looping)
+ * when dealing with larger objects.
+ *
+ * Usage:
+ *      const [ findPropVal, findMetaVal ] = fastSearch(authenticator);
+ *      i.e., findPropVal<string>({ defaultValue: "SHA1", key: "DigestAlgorithm" }),
+ *      i.e., findMetaVal<number>({ key: "displayOrder" }),
+ *
+ * @param authenticator
+ */
+export const fastSearch = (
+    authenticator: FederatedAuthenticatorWithMetaInterface
+): [ FindPropValFunction, FindMetaFunction ] => {
+
+    const propertyMap = new Map<string, any>();
+    authenticator.data?.properties.forEach(({ key, value }) => propertyMap.set(key, value));
+
+    const metadataMap = new Map<string, CommonPluggableComponentMetaPropertyInterface>();
+    authenticator.meta?.properties.forEach((meta) => metadataMap.set(meta.key, meta));
+
+    return [
+        <T>({ key, defaultValue }: FindPropValArgs<T>): T => {
+            if (propertyMap.has(key)) {
+                const value = propertyMap.get(key);
+                if (booleanSentAsAStringValue(value)) {
+                    return (castToBool(value) as unknown) as T;
+                }
+                return value;
+            }
+            return defaultValue;
+        },
+        ({ key }: FindMetaArgs) => {
+            if (metadataMap.has(key)) {
+                return metadataMap.get(key);
+            }
+            return null;
+        }
+    ];
+
+};
+
+/**
+ * For some reason the metadata API returns strings instead
+ * booleans for boolean default values. ¯\_(ツ)_/¯ Its good
```

Comment:
Please cleanup emojis from comments since it's not a standard practice

Reply:
Yup. I've removed this in commit b49a285